### PR TITLE
Add support for field and header prefix

### PR DESCRIPTION
**Pull request by rgannu**
DBZ-2504: Added support for field and header prefix.

### CODE REVIEW

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentState.java [hunk 1]
```diff
@@ -5,6 +5,19 @@
  */
 package io.debezium.connector.mongodb.transforms;
 
+import io.debezium.config.Configuration;
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentState.java:8]: Seems like you applied some auto-organization of imports? Can you please revert this, to keep the diff focused on the actual changes? You can run `mvn process-sources` to ensure the formatting is right before committing. Thx!

**Review by gunnarmorling — CHANGES_REQUESTED** [review]
Hey @rgannu, thx for this PR, but it's really hard to tell what the actual changes are with all the stylistic changes done at the same time. Please run the formatter (see inline comment), rework the commit and force-push. Thanks!

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentState.java (hunk 1 above)
Comment (rgannu) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentState.java]: yes.. My intellij reformat is doing that.. Fixed it now. Thanks

Comment (gunnarmorling) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentState.java]: No worries; looking much better now, thx!

**Review by gunnarmorling — COMMENTED** [review]
Thanks for the update, LGTM overall now, @rgannu! Could you only

* Add a test for the MongoDB implementation
* Add the new option to the docs under _documentation/modules/ROOT/pages/configuration/event-flattening.adoc_ and _documentation/modules/ROOT/pages/configuration/mongodb-event-flattening.adoc_, respectively

Once that's done, this is good to go. Thanks again!

**Comment by rgannu** [comment]
> Thanks for the update, LGTM overall now, @rgannu! Could you only
> 
> * Add a test for the MongoDB implementation
> * Add the new option to the docs under _documentation/modules/ROOT/pages/configuration/event-flattening.adoc_ and _documentation/modules/ROOT/pages/configuration/mongodb-event-flattening.adoc_, respectively
> 
> Once that's done, this is good to go. Thanks again!

Updated the documentation and added test in the MongoDB implementation.

**Comment by gunnarmorling** [comment]
Test failures are known and unrelated (tracked via https://issues.redhat.com/browse/DBZ-2268).

**Comment by gunnarmorling** [comment]
Rebased and applied. Great work, thanks a lot, @rgannu!

**Pull request merged**